### PR TITLE
remove out of date fields from DNS Policy API

### PR DESCRIPTION
**Pull request by philbrookes**

### CODE REVIEW

File: api/v1alpha1/dnspolicy_types.go [hunk 1]
```diff
@@ -239,33 +234,17 @@ func (l *DNSPolicyList) GetItems() []kuadrant.Policy {
...
-
-	return nil
+	// +kubebuilder:validation:Required
+	Endpoint string `json:"endpoint"`
+	// +kubebuilder:validation:Required
+	Port *int `json:"port"`
+	// +kubebuilder:validation:Required
+	Protocol *string `json:"protocol"`
+	// +kubebuilder:validation:Required
+	FailureThreshold *int `json:"failureThreshold"`
```
Comment (mikenairn) [inline api/v1alpha1/dnspolicy_types.go:244]: Should probably avoid using pointers if its required. Also, you don't need these lines `+kubebuilder:validation:Required` see https://github.com/Kuadrant/kuadrant-operator/pull/578

File: api/v1alpha1/dnspolicy_types.go [hunk 2]
```diff
@@ -239,37 +226,10 @@ func (l *DNSPolicyList) GetItems() []kuadrant.Policy {
...
-	}
-
-	if s.Protocol == nil {
-		protocol := "HTTPS"
-		s.Protocol = &protocol
-	}
+	Endpoint         string `json:"endpoint"`
+	Port             int    `json:"port"`
+	Protocol         string `json:"protocol"`
+	FailureThreshold int    `json:"failureThreshold"`
```
Comment (mikenairn) [inline api/v1alpha1/dnspolicy_types.go:232]: You might want to consider adding some CEL validations for these, presumably there are values that we should just not accept for some of these.

Comment (mikenairn) [reply api/v1alpha1/dnspolicy_types.go:232]: Also, a description of each would be good since it will end up as the description in the generated CRD.

**Review by mikenairn — APPROVED** [review]
Looks fine. Thought these were removed long ago tbh. Couple of suggestions, but nothing that need hold up the merge.

File: api/v1alpha1/dnspolicy_types.go (hunk 2 above)
Comment (philbrookes) [reply api/v1alpha1/dnspolicy_types.go]: nice, added.

File: api/v1alpha1/dnspolicy_types.go [hunk 3]
```diff
@@ -239,37 +226,20 @@ func (l *DNSPolicyList) GetItems() []kuadrant.Policy {
...
+	// Port to connect to the host on
+	// +kubebuilder:validation:Minimum:=1
+	Port int `json:"port"`
+	// Protocol to use when connecting to the host, valid values are "HTTP" or "HTTPS"
+	// +kubebuilder:validation:Enum:=HTTP;HTTPS
+	Protocol string `json:"protocol"`
+	// FailureThreshold is a limit of consecutive failures that must occur for a host
+	// to be considered unhealthy
+	// +kubebuilder:validation:Minimum:=1
+	FailureThreshold int `json:"failureThreshold"`
```
Comment (mikenairn) [inline api/v1alpha1/dnspolicy_types.go:242]: :+1:

**Review by mikenairn — APPROVED** [review]

**Comment by maleck13** [comment]
@philbrookes you need to regen the bundle also we will want to ensure docs call out the removal of these fields as this is a breaking change from the previous release 0.6.x

**Comment by philbrookes** [comment]
> @philbrookes you need to regen the bundle also we will want to ensure docs call out the removal of these fields as this is a breaking change from the previous release 0.6.x

yes, I'm working on a docs update already in a separate branch. AFAIK I have regenerated the bundle, is something missing?

**Pull request merged**